### PR TITLE
Update CutMix to use the BaseImageAugmentationLayer as base.

### PR DESCRIPTION
**Pull request by qlzh727**
I have to override the _augment_batch to maximize leverage of the existing code.

### CODE REVIEW

**Comment by qlzh727** [comment]
Update for https://github.com/keras-team/keras-cv/issues/202

**Comment by LukeWood** [comment]
Thanks for the PR @qlzh727!  Super close to v1 which is super exciting.  Can you add a test case to `xkeras_cv/layers/preprocessing/with_labels_test.py`?

File: keras_cv/layers/preprocessing/cut_mix.py [hunk 1]
```diff
@@ -48,23 +47,14 @@ def _sample_from_beta(alpha, beta, shape):
...
-        # pylint: disable=g-long-lambda
-        cutmix_augment = lambda: self._update_labels(*self._cutmix(images, labels))
-        no_augment = lambda: (images, labels)
-        return tf.cond(tf.cast(training, tf.bool), cutmix_augment, no_augment)
+    def _batch_augment(self, inputs):
+        images = inputs.get('images', None)
+        labels = inputs.get('labels', None)
+        if images is None or labels is None:
+            raise ValueError(
+                'CutMix is expecting both images and labels as inputs.'
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/cut_mix.py:55]: nit:  f"CutMix expects inputs in a dictionary with format {"images": images, "labels": labels}.

Got: inputs={inputs}"

This is more descriptive of what users should pass

File: keras_cv/layers/preprocessing/cut_mix.py [hunk 2]
```diff
@@ -12,13 +12,12 @@
...
 # limitations under the License.
 import tensorflow as tf
-import tensorflow.keras.layers as layers
 
 from keras_cv.utils import fill_utils
 
 
 @tf.keras.utils.register_keras_serializable(package="keras_cv")
-class CutMix(layers.Layer):
+class CutMix(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/cut_mix.py:20]: Should we override _augment and tf.Assert(False)?  This layer should never be called with 1 image.

Comment (LukeWood) [reply keras_cv/layers/preprocessing/cut_mix.py:20]: This ensures we only use _batch_augment

File: keras_cv/layers/preprocessing/cut_mix.py (hunk 1 above)
Comment (qlzh727) [reply keras_cv/layers/preprocessing/cut_mix.py]: Done.

**Comment by qlzh727** [comment]
> Thanks for the PR @qlzh727! Super close to v1 which is super exciting. Can you add a test case to `xkeras_cv/layers/preprocessing/with_labels_test.py`?

What is with_labels_test? I didn't find that in the current code.

**Comment by LukeWood** [comment]
> > Thanks for the PR @qlzh727! Super close to v1 which is super exciting. Can you add a test case to `xkeras_cv/layers/preprocessing/with_labels_test.py`?
> 
> What is with_labels_test? I didn't find that in the current code.

Ah sorry - it is in PR https://github.com/keras-team/keras-cv/pull/240

I'll merge this, rebase on top and add a example for CutMix.  Thanks!

**Comment by qlzh727** [comment]
Ack, will update the mixup in a next PR.
